### ISSUE TITLE
UX EPIC - Better date/time pickers

### ISSUE THREAD

**Issue by pconrad** [ucsb-cs156/proj-gauchoride#118]
The date/time pickers throughout Gauchoride (which are the default ones) are currently... not great.  For example in some cases 

In this issue, you'll first create three new internal components:
* `<GauchorideDateTimePicker>`
* `<GauchorideDatePicker>` 
* `<GauchorideTimePicker>` 

Initially (first issue): these will just be wrappers around the *existing techniques* for picking dates and times.  The idea is to first create (could be three separate issues) new React Components in a "frontend, new component only" story. Those are relatively easy stories to implement and get approved.

Then, once any one of these is merged into main, folks on the team can divide up the pages and start replacing the hard coded date pickers, date time pickers, and time pickers with these named components. Initally there's no change in the UX.

But then, once all date/time picking in the app is using the  `<GauchorideDateTimePicker>`, if we decide at a later stage to switch to a different implementation of the date/time picker, we only have to change it in one place in the app, and it's changed everywhere.

And: in parallel, team members can research and experiment with different implementations of each of these, using a variety of components (see options below).  The idea is to choose ones that provide a good user experience, but are also well maintained, documented, and easy for devs to work with.

The team could create multiple PRs with different options, and then review those in Storybook on the PR branches to decide which one they prefer, before picking a winner to merge.   

# Options for the new date/time picker.

A few options:
* https://www.npmjs.com/package/@mui/x-date-pickers
* Any of the choices here: https://www.npmjs.com/search?q=time%20picker

When selecting, keep in mind not only the look and ease of use of the date/time picker, but also:
* How long has the project been around?  A few years, or only a few months or days?
* How recently has it been updated?  Within the last few months? Or was the last update years ago?
* How actively does the project seem to be maintained?

The point is that you want to avoid creating a dependency on a project that is unlikely to get updates as node, react, react-bootstrap, etc. continue to change and evolve.

# A1: Frontend. Create `<GauchorideDateTimePicker>`

Acceptance Criteria

- [ ] There is a component called  `<GauchorideDateTimePicker>` with a storybook story and tests
- [ ] There is at least a draft (separate) PR as a proof of concept that shows how it would be integrated into one of the existing page in the Gauchoride app and used in place of the current  `<GauchorideDateTimePicker>`  (this is just to show that the API is reasonable.

# A2: Frontend, Create `<GauchorideDatePicker>`

Acceptance Criteria

- [ ] There is a component called  `<GauchorideDatePicker>` with a storybook story and tests
- [ ] There is at least a draft (separate) PR as a proof of concept that shows how it would be integrated into one of the existing page in the Gauchoride app and used in place of the current  `<GauchorideDatePicker>`  (this is just to show that the API is reasonable.

# A3:  Frontend, Create `<GauchorideTimePicker>`

Acceptance Criteria

- [ ] There is a component called  `<GauchorideTimePicker>` with a storybook story and tests
- [ ] There is at least a draft (separate) PR as a proof of concept that shows how it would be integrated into one of the existing page in the Gauchoride app and used in place of the current  `<GauchorideTimePicker>`  (this is just to show that the API is reasonable.

# B1a, B1b, B1c, etc.: Frontend : Replace old date/time data entry with `<GauchorideDateTimePicker>` on each applicable component.

Work one form / page at a time, finding data entry for date/times and replacing it with `<GauchorideDateTimePicker>`

Each of the B1a, B1b, B1c would be one component or page in the application where the new component has been introduced, and each would be merged to main separately (small focused PRs.)  It may be ok to combine some, but keeping PRs small and focused is the idea.

Acceptance Criteria

- [ ] The form now uses the  `<GauchorideDateTimePicker>` instead of the old way.

# B2a, B2b, B2c, etc.: Frontend : Replace old date/time data entry with `<GauchorideDatePicker>` on each applicable component.

Work one form / page at a time, finding data entry for date/times and replacing it with `<GauchorideDatePicker>`

Each of the B2a, B2b, B2c would be one component or page in the application where the new component has been introduced, and each would be merged to main separately (small focused PRs.)  It may be ok to combine some, but keeping PRs small and focused is the idea.

Acceptance Criteria

- [ ] The form now uses the  `<GauchorideDatePicker>` instead of the old way.

# B3a, B3b, B3c, etc.: Frontend : Replace old date/time data entry with `<GauchorideTimePicker>` on each applicable component.   

Work one form / page at a time, finding data entry for date/times and replacing it with `<GauchorideTimePicker>`

Each of the B3a, B3b, B3c would be one component or page in the application where the new component has been introduced, and each would be merged to main separately (small focused PRs.)  It may be ok to combine some, but keeping PRs small and focused is the idea.

Acceptance Criteria

- [ ] The form now uses the  `<GauchorideTimePicker>` instead of the old way.

# C1a, C1b, C1c, etc.: Frontend : New version of`<GauchorideDateTimePicker>` using various libraries, modules, pre-defined date/time pickers.

The idea here of C1a, C1b, C1c is that they might use different libraries to try out various ways of implementing the picker, but *only one would be merged*.

Acceptance Criteria
 
- [ ] The component has an improved UX (team members compare old and new implementation and the consensus is that the new one is better.)
- [ ] Any new dependencies added to `package.json` are from well supported libraries (document this in the PR description)
- [ ] The new component functions properly everywhere it is used in the application.
- [ ] Ideally, we would allow prospective Gauchoride users to actually make the final selection.

# C2a, C2b, C2c, etc.: Frontend : New version of`<GauchorideDatePicker>` using various libraries, modules, pre-defined date/time pickers.

The idea here of C2a, C2b, C2c is that they might use different libraries to try out various ways of implementing the picker, but *only one would be merged*.

Acceptance Criteria
 
- [ ] The component has an improved UX (team members compare old and new implementation and the consensus is that the new one is better.)
- [ ] Any new dependencies added to `package.json` are from well supported libraries (document this in the PR description)
- [ ] The new component functions properly everywhere it is used in the application.
- [ ] Ideally, we would allow prospective Gauchoride users to actually make the final selection.

# C3a, C3b, C3c, etc.: Frontend : New version of`<GauchorideTimePicker>` using various libraries, modules, pre-defined date/time pickers.

The idea here of C3a, C3b, C3c is that they might use different libraries to try out various ways of implementing the picker, but *only one would be merged*.

Acceptance Criteria
 
- [ ] The component has an improved UX (team members compare old and new implementation and the consensus is that the new one is better.)
- [ ] Any new dependencies added to `package.json` are from well supported libraries (document this in the PR description)
- [ ] The new component functions properly everywhere it is used in the application.
- [ ] Ideally, we would allow prospective Gauchoride users to actually make the final selection.
